x-pack/plugin/esql/qa/testFixtures/src/main/resources/match-function.csv-spec

```diff
@@ -197,3 +197,253 @@ emp_no:integer | first_name:keyword | last_name:keyword
 10041          | Uri                 | Lenart         
 10043          | Yishay              | Tzvieli        
 ;
+
+testMatchIpField
+required_capability: match_function
+required_capability: match_additional_types
+
+from sample_data
+| where match(client_ip, "172.21.0.5"::IP)
+| keep client_ip, message;
+
+client_ip:ip   | message:keyword
+172.21.0.5     | Disconnected
+;
+
+testMatchDateField
+required_capability: match_function
+required_capability: match_additional_types
+
+from date_nanos
+| where match(millis, "2023-10-23T13:55:01.543Z"::DATETIME)
+| keep millis;
+
+millis:date
+2023-10-23T13:55:01.543Z
+;
+
+testMatchDateNanosField
+required_capability: match_function
+required_capability: match_additional_types
+
+from date_nanos
+| where match(nanos, "2023-10-23T13:55:01.543123456Z"::DATE_NANOS)
+| keep nanos;
+
+nanos:date_nanos
+2023-10-23T13:55:01.543123456Z
+;
+
+testMatchBooleanField
+required_capability: match_function
+required_capability: match_additional_types
+
+from employees 
+| where match(still_hired, true) and height > 2.08 
+| keep first_name, still_hired, height;
+ignoreOrder:true
+
+first_name:keyword  | still_hired:boolean   | height:double
+Saniya              | true                  | 2.1
+Yongqiao            | true                  | 2.1
+Kwee                | true                  | 2.1
+Amabile             | true                  | 2.09
+;
+
+testMatchIntegerField
+required_capability: match_function
+required_capability: match_additional_types
+
+from employees 
+| where match(emp_no, 10004)
+| keep emp_no, first_name;
+
+emp_no:integer | first_name:keyword
+10004          | Chirstian
+;
+
+testMatchDoubleField
+required_capability: match_function
+required_capability: match_additional_types
+
+from employees 
+| where match(salary_change, 9.07)
+| keep emp_no, salary_change;
+
+emp_no:integer | salary_change:double
+10014          | [-1.89, 9.07] 
+;  
+
+testMatchLongField
+required_capability: match_function
+required_capability: match_additional_types
+
+from date_nanos
+| where match(num, 1698069301543123456)
+| keep num;
+
+num:long
+1698069301543123456
+;
+
+testMatchUnsignedLongField
+required_capability: match_function
+required_capability: match_additional_types
+
+from ul_logs
+| where match(bytes_out, "12749081495402663265"::UNSIGNED_LONG)
+| keep bytes_out;
+
+bytes_out:unsigned_long
+12749081495402663265
+;
+
+testMatchVersionField
+required_capability: match_function
+required_capability: match_additional_types
+
+from apps 
+| where match(version, "2.1"::VERSION) 
+| keep name, version;
+
+name:keyword | version:version
+bbbbb        | 2.1
+;
+
+testMatchIpFieldAsString
```

Comment:
The following tests use strings, which always do the right thing in `match` in terms of type.